File: Entities/IEntityManager.vb
Imports Entities

Public Interface IEntityManager
    Sub RegisterEntity(ByRef newEntity As BaseGameEntity)
    Sub RemoveEntity(ByRef pEntity As BaseGameEntity)
    Sub UpdateAll()
    Function GetEntityFromId(id As Guid) As BaseGameEntity
    Function GetAllEntities() As List(Of BaseGameEntity)
End Interface

File: space-game/Entities/IEntityManager.cs
using System;
using System.Collections.Generic;

namespace Entities
{
    public interface IEntityManager
    {
        void RegisterEntity(BaseGameEntity newEntity);
        void RemoveEntity(BaseGameEntity pEntity);
        void UpdateAll();
        BaseGameEntity GetEntityFromId(Guid id);
        List<BaseGameEntity> GetAllEntities();
    }
}
